Can't clone wiki

<!--
    1. Please speak English, this is the language all of us can speak and write.
    2. Please ask questions or configuration/deploy problems on our Discord 
       server (https://discord.gg/NsatcWJ) or forum (https://discourse.gitea.io).
    3. Please take a moment to check that your issue doesn't already exist.
    4. Please give all relevant information below for bug reports, because 
       incomplete details will be handled as an invalid report.
-->

- Gitea version (or commit ref): 1.5.3
- Git version: 2.19.1
- Operating system: archlinux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
  - [x] Not gonna creating an account (sorry)
- Log gist:

## Description
I'm unable to clone a projects wiki. Cloning the repo itself works just fine though.
Oh and, I'm using the official docker image.

git client error:
```
$ git clone ssh://git@<host>:23/tadly/test.wiki.git
Cloning into 'test.wiki'...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x8d7b78]

goroutine 1 [running]:
code.gitea.io/gitea/vendor/github.com/go-xorm/xorm.(*Engine).Quote(0x0, 0x15731e6, 0x4, 0x1, 0xc00057c570)
        /go/src/code.gitea.io/gitea/vendor/github.com/go-xorm/xorm/engine.go:187 +0x78
code.gitea.io/gitea/vendor/github.com/go-xorm/xorm.(*Statement).colmap2NewColsWithQuote(0xc0000c4518, 0x15731e6, 0x4, 0xc00057c501)
        /go/src/code.gitea.io/gitea/vendor/github.com/go-xorm/xorm/statement.go:602 +0x121
code.gitea.io/gitea/vendor/github.com/go-xorm/xorm.(*Statement).Cols(0xc0000c4518, 0xc0005bb320, 0x1, 0x1, 0xc0000c4500)
        /go/src/code.gitea.io/gitea/vendor/github.com/go-xorm/xorm/statement.go:633 +0xc2
code.gitea.io/gitea/vendor/github.com/go-xorm/xorm.(*Session).Cols(0xc0000c4500, 0xc0005bb320, 0x1, 0x1, 0x18)
        /go/src/code.gitea.io/gitea/vendor/github.com/go-xorm/xorm/session_cols.go:148 +0x4f
code.gitea.io/gitea/models.(*Repository).getOwnerName(0xc000087980, 0x173f320, 0x0, 0xc0000de4e0, 0x1ed)
        /go/src/code.gitea.io/gitea/models/repo.go:502 +0x136
code.gitea.io/gitea/models.(*Repository).mustOwnerName(0xc000087980, 0x173f320, 0x0, 0x0, 0x0)
        /go/src/code.gitea.io/gitea/models/repo.go:518 +0x43
code.gitea.io/gitea/models.(*Repository).MustOwnerName(0xc000087980, 0x1, 0x4)
        /go/src/code.gitea.io/gitea/models/repo.go:234 +0x43
code.gitea.io/gitea/models.(*Repository).WikiPath(0xc000087980, 0xf, 0xc0005bb4a8)
        /go/src/code.gitea.io/gitea/models/wiki.go:70 +0x2b
code.gitea.io/gitea/models.(*Repository).HasWiki(0xc000087980, 0xc00058a520)
        /go/src/code.gitea.io/gitea/models/wiki.go:75 +0x2b
code.gitea.io/gitea/models.(*Repository).InitWiki(0xc000087980, 0xf, 0xc0005bb6d0)
        /go/src/code.gitea.io/gitea/models/wiki.go:81 +0x2f
code.gitea.io/gitea/cmd.runServ(0xc0000fd2c0, 0x0, 0x0)
        /go/src/code.gitea.io/gitea/cmd/serv.go:317 +0xd0f
code.gitea.io/gitea/vendor/github.com/urfave/cli.HandleAction(0x13cc700, 0x15eb488, 0xc0000fd2c0, 0xc000078700, 0x0)
        /go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:471 +0xad
code.gitea.io/gitea/vendor/github.com/urfave/cli.Command.Run(0x1573376, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x15c9a64, 0x2f, 0x0, ...)
        /go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/command.go:191 +0x96d
code.gitea.io/gitea/vendor/github.com/urfave/cli.(*App).Run(0xc000128820, 0xc0000300c0, 0x4, 0x4, 0x0, 0x0)
        /go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:241 +0x5a2
main.main()
        /go/src/code.gitea.io/gitea/main.go:56 +0x42f
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

--- a/cmd/serv.go
+++ b/cmd/serv.go
@@ -144,11 +144,15 @@ func runServ(c *cli.Context) error {
 		}()
 	}
 
-	isWiki := false
-	unitType := models.UnitTypeCode
+	var (
+		isWiki   bool
+		unitType = models.UnitTypeCode
+		unitName = "code"
+	)
 	if strings.HasSuffix(reponame, ".wiki") {
 		isWiki = true
 		unitType = models.UnitTypeWiki
+		unitName = "wiki"
 		reponame = reponame[:len(reponame)-5]
 	}
 
@@ -245,7 +249,7 @@ func runServ(c *cli.Context) error {
 					clientMessage = "You do not have sufficient authorization for this action"
 				}
 				fail(clientMessage,
-					"User %s does not have level %v access to repository %s",
+					"User %s does not have level %v access to repository %s's "+unitName,
 					user.Name, requestedMode, repoPath)
 			}
 
@@ -304,7 +308,7 @@ func runServ(c *cli.Context) error {
 		gitcmd = exec.Command(verb, repoPath)
 	}
 	if isWiki {
-		if err = repo.InitWiki(); err != nil {
+		if err = private.InitWiki(repo.ID); err != nil {
 			fail("Internal error", "Failed to init wiki repo: %v", err)
 		}
 	}

--- a/modules/private/wiki.go
+++ b/modules/private/wiki.go
@@ -1,0 +1,33 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package private
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// InitWiki initwiki via repo id
+func InitWiki(repoID int64) error {
+	// Ask for running deliver hook and test pull request tasks.
+	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/repositories/%d/wiki/init", repoID)
+	log.GitLogger.Trace("InitWiki: %s", reqURL)
+
+	resp, err := newInternalRequest(reqURL, "GET").Response()
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	// All 2XX status codes are accepted and others will return an error
+	if resp.StatusCode/100 != 2 {
+		return fmt.Errorf("Failed to init wiki: %s", decodeJSONError(resp).Err)
+	}
+
+	return nil
+}

--- a/routers/private/internal.go
+++ b/routers/private/internal.go
@@ -82,6 +82,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 		m.Post("/repositories/:repoid/keys/:keyid/update", UpdateDeployKey)
 		m.Get("/repositories/:repoid/user/:userid/checkunituser", CheckUnitUser)
 		m.Get("/repositories/:repoid/has-keys/:keyid", HasDeployKey)
+		m.Get("/repositories/:repoid/wiki/init", InitWiki)
 		m.Post("/push/update", PushUpdate)
 		m.Get("/protectedbranch/:pbid/:userid", CanUserPush)
 		m.Get("/repo/:owner/:repo", GetRepositoryByOwnerAndName)

--- a/routers/private/wiki.go
+++ b/routers/private/wiki.go
@@ -1,0 +1,34 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package private
+
+import (
+	"code.gitea.io/gitea/models"
+
+	macaron "gopkg.in/macaron.v1"
+)
+
+// InitWiki initilizes wiki via repo id
+func InitWiki(ctx *macaron.Context) {
+	repoID := ctx.ParamsInt64("repoid")
+
+	repo, err := models.GetRepositoryByID(repoID)
+	if err != nil {
+		ctx.JSON(500, map[string]interface{}{
+			"err": err.Error(),
+		})
+		return
+	}
+
+	err = repo.InitWiki()
+	if err != nil {
+		ctx.JSON(500, map[string]interface{}{
+			"err": err.Error(),
+		})
+		return
+	}
+
+	ctx.Status(202)
+}

